### ISSUE TITLE
prefer-rest-params warns when using arguments.callee.name

### ISSUE THREAD

**Issue by krainboltgreene** [eslint/eslint#6835]
**What version of ESLint are you using?** 2

**What parser (default, Babel-ESLint, etc.) are you using?** babel

**Please show your full configuration:** It's really long.

**What did you do? Please include the actual source code causing the issue.** I wrote the code:

```
export default function freeMemory () {
  return this.store.hset(arguments.callee.name, new Date(), os.freeMemory())
}
```

**What did you expect to happen?**

Nothing.

**What actually happened? Please include the actual, raw output from ESLint.**

I got a linter error from `prefer-rest-params`, because I was using arguments. I don't believe there's a way to write what I did without calling arguments.

**Comment by vitorbal** [comment]
Hey @krainboltgreene, thanks for reaching out!

@mysticatea this seems similar to #5990, as in, it would also be a relaxing enhancement if accepted. What do you think?

**Comment by krainboltgreene** [comment]
100%, that seems absolutely right. Should I close?

**Comment by vitorbal** [comment]
@krainboltgreene Sorry if I wasn't clear, I think this is another false-positive that needs to be taken care of, #5990 deals with a different false-positive, so we should keep this issue open.

Just wanted to loop @mysticatea in since he is championing #5990 and probably has an opinion on this issue as well.

**Comment by nzakas** [comment]
@krainboltgreene is that the complete source code you used or was that Ina function? (We need a complete example to verify.) Also, please provide the actual ESLint output, not just a description. Thanks!

**Comment by krainboltgreene** [comment]
Updated with actual code, but I believe using that anywhere would get you the linter error.

**Comment by krainboltgreene** [comment]
Looks like the `no-caller` has a similar situation, where it's thinking `arguments.callee.name` is similar to calling the function (like `arguments.callee()`).

**Comment by mysticatea** [comment]
Yes, this is very similar to #5990.
I will make the rule ignoring `arguments` if it's to access the properties except numbers.
So, #5990 would fix this issue as well.

```js
// ✘
arguments[1]    // ← a property access to a number.
arguments[v]    // ← a property access to a number possibly.
arguments       // ← a use of whole arguments.

// ✔ GOOD
arguments.length
arguments.callee
arguments.foo
```

**Comment by vitorbal** [comment]
@mysticatea thanks for the update! In that case, I think we can indeed close this issue in favor of #5990.

@krainboltgreene please follow #5990 for updates. Thanks for the report and your support!

Closing this, please re-open if anyone disagrees.
